content/blog/2018/01/2018-01-06-jep-200.adoc

```diff
@@ -0,0 +1,79 @@
+---
+layout: post
+title: "JEP-200: Remoting / XStream whitelist integrated into Jenkins core"
+tags:
+- core
+- security
+- remoting
+author: jglick
+---
+
+link:https://github.com/jenkinsci/jep/blob/master/jep/200/README.adoc[JEP-200] has been integrated into Jenkins weekly builds
+and (if all goes well) will be a part of the next LTS line.
+In a nutshell, this change is a security hardening measure
+to be less permissive about deserializing Java classes defined in the Java Platform or libraries bundled with Jenkins.
+For several years now, Jenkins has specifically _blacklisted_ certain classes and packages according to known or suspected exploits;
+now it will reject all classes not explicitly mentioned in a _whitelist_, or defined in Jenkins core or plugins.
+
+To the extent that advance testing of the impact of this change on popular plugins has been completed,
+most users (and even plugin developers) should not notice any difference.
+If you do encounter a `java.lang.SecurityException: Rejected: some.pkg.and.ClassName` in the Jenkins UI or logs,
+you may have found a case where an unusual plugin, or an unusual usage mode of a common plugin,
+violates the existing whitelist.
+This will be visible in the Jenkins system log as a message from `jenkins.security.ClassFilterImpl` like the following:
+
+----
+some.pkg.and.ClassName in file:/var/lib/jenkins/plugins/some-plugin-name/WEB-INF/lib/some-library-1.2.jar might be dangerous, so rejecting; see https://jenkins.io/redirect/class-filter/
+----
+
+where the link would direct you here.
+
+If you find such a case, please report it in the Jenkins issue tracker, under the appropriate plugin component.
+Link it to link:https://issues.jenkins-ci.org/browse/JENKINS-47736[JENKINS-47736] and add the `JEP-200` label.
+If at all possible, include complete steps to reproduce the problem from scratch.
+Jenkins developers will strive to evaluate the reason for the violation and offer a fix in the form of a core and/or plugin update.
+
+Assuming you see no particular reason to think that the class in question has dangerous deserialization semantics, which is rare,
+it is possible to work around the problem in your own installation as a temporary expedient.
+Simply make note of any class name(s) mentioned in such log messages,
+and run Jenkins with this startup option (details will depend on your installation method):
+
+----
+-Dhudson.remoting.ClassFilter=some.pkg.and.ClassName,some.pkg.and.OtherClassName
+----
+
+As a plugin developer encountering this kind of error,
+your first task is to ensure that it is reproducible in a functional (`JenkinsRule`) test
+when running
+
+[source,sh]
+----
+mvn test -Djenkins.version=2.105
```

Comment:
I would be explicit that it requires the Parent POM 2.0+ . Referencing Dockerized PCT could help in general, but there are still some bugs in the PCT logic (though manual run may be non-trivial as well: detached plugins & Co)

Reply:
Will expand a bit.

Reply:
Needs to be replaced by 2.102